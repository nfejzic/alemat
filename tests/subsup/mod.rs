use alemat::{
    elements::{scripted::SubSup, Ident, Num, Operator},
    MathMl, MathMlAttr, MathMlFormatter,
};

#[test]
fn subsup() {
    let out = MathMl::with_content(
        SubSup::builder()
            .subscript(Ident::from("0"))
            .supscript(Ident::from("1"))
            .base(Ident::from("x"))
            .build(),
    )
    .render(&mut MathMlFormatter);

    crate::snap_test!(out, name: "subsup_subsup");
}

#[test]
fn integral() {
    let out = MathMl::with_content(alemat::children![
        SubSup::builder()
            .base(Operator::integral())
            .subscript(Ident::from("0"))
            .supscript(Ident::from("1"))
            .build(),
        Ident::from("x"),
        Ident::from("dx"),
    ])
    .with_attr([MathMlAttr::Display(String::from("block"))])
    .render(&mut MathMlFormatter);

    crate::snap_test!(out, name: "subsup_integral");
}

#[test]
fn summation() {
    let out = MathMl::with_content(alemat::children![
        SubSup::builder()
            .base(Operator::sum())
            .subscript(alemat::row![Ident::from("i"), Operator::eq(), Num::from(0)])
            .supscript(Ident::from("k"))
            .build(),
        SubSup::builder()
            .base(Num::from(2))
            .supscript(Ident::from("i"))
            .build()
    ])
    .with_attr([MathMlAttr::Display(String::from("block"))])
    .render(&mut MathMlFormatter);

    crate::snap_test!(out, name: "subsup_summation");
}
