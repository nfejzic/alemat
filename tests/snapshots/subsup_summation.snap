---
source: tests/subsup/mod.rs
expression: out
---
<math display="block"><msubsup ><mo >âˆ‘</mo><mrow ><mi >i</mi><mo >=</mo><mn >0</mn></mrow><mi >k</mi></msubsup><msup ><mn >2</mn><mi >i</mi></msup></math>
